src/models/OAuthModel.php

```diff
@@ -125,16 +125,22 @@ public function createAccessTokenFromPassword($clientId, $username, $password)
      */
     protected function getUserId($username, $password)
     {
-        $sql  = 'SELECT ID, password, email FROM user
-            WHERE username=:username
-            AND verified = 1';
+        $sql  = 'SELECT ID, password, email, verified FROM user
+            WHERE username=:username';
         $stmt = $this->_db->prepare($sql);
         $stmt->execute(array("username" => $username));
         $result = $stmt->fetch(PDO::FETCH_ASSOC);
+
         if ($result) {
-            if (password_verify(md5($password), $result['password'])) {
```

Comment:
While you're at it, would it make sense to fail early and rewrite that whole conditional tree?

```php
if (! $result) return false;
if ($result['verified'] !== 1) return false;
// etc.
```